### PR TITLE
story 905

### PR DESCRIPTION
**Pull request by rahulramakrishnan3**

### CODE REVIEW

File: src/app/core/services/form/form.service.ts [hunk 1]
```diff
@@ -7,7 +7,9 @@ import * as _ from 'lodash-es';
   providedIn: 'root',
 })
 export class FormService {
-  constructor(private http: HttpService, private db: DbService) { }
+  constructor(private http: HttpService, private db: DbService,
+   private httpService: HttpService
```
Comment (Cafnanc) [inline src/app/core/services/form/form.service.ts]: Why http two times injected?

File: src/app/core/services/permission/permission.service.ts [hunk 2]
```diff
@@ -54,6 +54,7 @@ export class PermissionService {
     try {
       const data: any = await this.httpService.get(config);
       this.setConfigInLocal(data.result)
+      return data
```
Comment (Cafnanc) [inline src/app/core/services/permission/permission.service.ts:57]: Why this change?

File: src/app/pages/home-search/home-search.page.ts [hunk 3]
```diff
@@ -1,121 +1,248 @@
...
-      case 'enrollAction':
-        let enrollResult = await this.sessionService.enrollSession(event.data.id);
-        if(enrollResult.result){
-          this.toast.showToast(enrollResult.message, "success")
-          this.search();
-        }
-        break;
+  closeCriteriaChip(){
+    this.criteriaChip = "";
+    this.criteriaChipName = "";
```
Comment (Cafnanc) [inline src/app/pages/home-search/home-search.page.ts]: why two variables? cant you use it as a single object with name and label if needed?

File: src/app/pages/home-search/home-search.page.ts [hunk 4]
```diff
@@ -1,121 +1,248 @@
...
+  locationBack(){
+    this.location.back()
   }
 
-  async onSessionAction(event){
-    switch (event.type) {
-      case 'cardSelect':
-        this.router.navigate([`/${CommonRoutes.SESSIONS_DETAILS}/${event.data.id}`])
-        break;
+  extractLabels(data) {
```
Comment (Cafnanc) [inline src/app/pages/home-search/home-search.page.ts:168]: do we need this? from filter itself u can get label and name right? why extra function

File: src/app/pages/mentor-search-directory/mentor-search-directory.page.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+          value: entity.value
+        })),
+        type: "checkbox"
+      };
+    });
+  
+    return filterData;
+  }
+
+  extractLabels(data) {
```
Comment (Cafnanc) [inline src/app/pages/mentor-search-directory/mentor-search-directory.page.ts:133]: is this needed?

File: src/app/pages/mentor-search-directory/mentor-search-directory.page.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    this.flatChips = [];
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        this.flatChips.push(...data[key]);
+      }
+    }
+  }
+
+  getFilteredData() {
```
Comment (Cafnanc) [inline src/app/pages/mentor-search-directory/mentor-search-directory.page.ts]: can u change all the name to a meaning full name? For eg: is this function giving filterData? no right. its creating a query string i think.

File: src/app/pages/mentor-search-directory/mentor-search-directory.page.ts [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+    const queryString = Object.keys(this.filteredDatas)
+      .map(key => `${key}=${this.filteredDatas[key]}`)
+      .join('&');
+
+    this.urlFilterData = queryString;
```
Comment (Cafnanc) [inline src/app/pages/mentor-search-directory/mentor-search-directory.page.ts]: can u return query string from here?

File: src/app/pages/mentor-search-directory/mentor-search-directory.page.ts [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    console.log(this.filteredDatas, this.urlFilterData)
+  }
+
+  async getMentors(showLoader = true){
```
Comment (Cafnanc) [inline src/app/pages/mentor-search-directory/mentor-search-directory.page.ts]: pls add api calls in corresponding service files

File: src/app/shared/components/filter-tree/filter-tree.component.scss [hunk 9]
```diff
@@ -5,16 +5,18 @@
 }
 .wrapper-class {
     display: flex;
-    justify-content: space-between;
+    justify-content: end;
+    padding: 5px 5px;
 }
 .underline {
-    border-bottom: 1px solid black;
+    // border-bottom: 1px solid black;
```
Comment (Cafnanc) [inline src/app/shared/components/filter-tree/filter-tree.component.scss]: remove commented lines

File: src/app/shared/components/filter-tree/filter-tree.component.html [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <ion-header *ngIf="enableFilterHeader && filterData">
-  <div class="ion-padding-start wrapper-class ion-padding-bottom">
-    <ion-label class="header">{{"FILTER"|translate}}</ion-label>
+  <div class="ion-padding-start wrapper-class">
+    <!-- <ion-label class="header">{{"FILTER"|translate}}</ion-label> -->
```
Comment (Cafnanc) [inline src/app/shared/components/filter-tree/filter-tree.component.html]: remove

File: src/app/pages/tabs/mentor-directory/mentor-directory.page.scss [hunk 11]
```diff
@@ -13,4 +13,91 @@
...
+}
+
+
+
+
+
+
+
+
+///////////////////////
```
Comment (Cafnanc) [inline src/app/pages/tabs/mentor-directory/mentor-directory.page.scss]: what is this?

File: src/app/core/services/permission/permission.service.ts (hunk 2 above)
Comment (rahulramakrishnan3) [reply src/app/core/services/permission/permission.service.ts:57]: criteria chip data is in this api. we need this in new pages
